### PR TITLE
feat(strongbox): create @appium/strongbox

### PR DESCRIPTION
**Pull request by boneskull**
This PR creates a new package `@appium/strongbox`, which provides a generic persistence store for Appium extensions.

A similar module has been used by several of HeadSpin's extensions, but it is time to move it into the Appium monorepo so we avoid more duplication.

See `README.md` for more details.

### CODE REVIEW

**Comment by boneskull** [comment]
Current dependencies on/for this PR:
* master
  * **PR #18471** <a href="https://app.graphite.dev/github/pr/appium/appium/18471" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/appium/appium/18471?utm_source=stack-comment).

File: packages/strongbox/lib/index.ts [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+   * Slugifies the name & determines the directory
+   * @param name Name of instance
+   * @param opts Options
+   */
+  protected constructor(
+    public readonly name: string,
+    {container, suffix = DEFAULT_SUFFIX, defaultCtor = BaseItem}: StrongboxOpts = {}
+  ) {
+    this.id = slugify(name);
+    this.defaultCtor = defaultCtor;
```
Comment (jlipps) [inline packages/strongbox/lib/index.ts:144]: maybe too wordy but `defaultItemCtor` might be more clear than `defaultCtor` (which could be taken to refer to the Strongbox constructor?)

**Review by jlipps — APPROVED** [review]

File: packages/strongbox/lib/index.ts (hunk 1 above)
Comment (boneskull) [reply packages/strongbox/lib/index.ts:144]: mm, yes.
